language/vm/vm-runtime/src/gas_meter.rs

```diff
@@ -23,11 +93,12 @@ pub struct GasMeter {
 // NB: A number of the functions/methods in this struct will return a VMResult<T>
 // since we will need to access stack and memory states, and we need to be able
 // to report errors properly from these accesses.
-impl GasMeter {
+impl<'txn> GasMeter<'txn> {
```

Comment:
I agree about the passing, but don't know about the lifetime -- the lifetime exists because we load the schedule for the block and then pass immutable references to this in to each transaction. 

With possible updates to the gas schedule, this might also go away -- each transaction could take ownership over and then return the updated gas schedule. But, that's for a separate PR.